h2/utilities.py

```diff
@@ -164,6 +169,12 @@ def validate_headers(headers):
     headers = _reject_te(headers)
     headers = _reject_connection_header(headers)
     headers = _reject_pseudo_header_fields(headers)
+
+    # We only expect to see :authority and Host headers on request header
+    # blocks, so only perform this validation if we're on the server side.
+    if (not client_side) and (not is_trailer):
+        headers = _check_host_authority_header(headers)
```

Comment:
So ideally we'd have a single state namedtuple here, and then each filter would accept both the headers and the state tuple as an argument. That lets us define essentially a single logical "filter function". The pre-existing filters just ignore the state, but then `_check_host_authority_header` can use it to make a decision (and in this case, decide to be a no-op).